src/transport/http2.rs

```diff
@@ -307,35 +317,157 @@ impl<In: RpcMessage, Out: RpcMessage> Clone for ServerChannel<In, Out> {
         Self {
             channel: self.channel.clone(),
             stop_tx: self.stop_tx.clone(),
+            config: self.config.clone(),
+            _p: PhantomData,
         }
     }
 }
 
 /// Receive stream for http2 channels.
-///
-/// This is just a mem channel that is fed from a task that reads from the hyper stream.
-pub type RecvStream<M> = super::mem::RecvStream<M>;
+pub struct RecvStream<Res: RpcMessage>(
+    flume::r#async::RecvStream<'static, hyper::Result<Bytes>>,
+    PhantomData<Res>,
+);
 
-/// Send sink for http2 channels.
-///
-/// This is just a mem channel that is read from a task that writes to the hyper stream.
-pub type SendSink<M> = super::mem::SendSink<M>;
+impl<Res: RpcMessage> RecvStream<Res> {
+    /// Creates a new [`RecvStream`] from a [`flume::Receiver`].
+    pub fn new(recv: flume::Receiver<hyper::Result<Bytes>>) -> Self {
+        Self(recv.into_stream(), PhantomData)
+    }
+}
+
+impl<In: RpcMessage> Clone for RecvStream<In> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone(), PhantomData)
+    }
+}
+
+impl<Res: RpcMessage> futures::Stream for RecvStream<Res> {
+    type Item = Result<Res, RecvError>;
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        match self.0.poll_next_unpin(cx) {
+            Poll::Ready(Some(Ok(item))) => match bincode::deserialize::<Res>(item.as_ref()) {
+                Ok(msg) => Poll::Ready(Some(Ok(msg))),
+                Err(cause) => Poll::Ready(Some(Err(RecvError::DeserializeError(cause)))),
+            },
+            Poll::Ready(Some(Err(cause))) => Poll::Ready(Some(Err(RecvError::NetworkError(cause)))),
+            Poll::Ready(None) => Poll::Ready(None),
+            Poll::Pending => Poll::Pending,
+        }
+    }
+}
+
+/// SendSink for http2 channels
+pub struct SendSink<Out: RpcMessage> {
+    sink: flume::r#async::SendSink<'static, io::Result<Bytes>>,
+    config: Arc<ChannelConfig>,
+    _p: PhantomData<Out>,
+}
+
+impl<Out: RpcMessage> SendSink<Out> {
+    fn new(sender: flume::Sender<io::Result<Bytes>>, config: Arc<ChannelConfig>) -> Self {
+        Self {
+            sink: sender.into_sink(),
+            config,
+            _p: PhantomData,
+        }
+    }
+    fn serialize(&self, item: Out) -> Result<Bytes, SendError> {
+        let data = bincode::serialize(&item).map_err(SendError::SerializeError)?;
+        let max_payload_size = self.config.max_frame_size as usize - 1024;
```

Comment:
What is this magic constant?  Might be worth explaining.

Reply:
It's a fudge factor. Probably on the high side. I will add a comment.